push.bash

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+fail() {
+  echo "::error::$1"
+  exit 1
+}
+
+# Make sure the token isn't accidentally logged
+echo "::add-mask::${BUF_TOKEN}"
+
+if [ -z "${GITHUB_SHA}" ]; then
+  fail "the commit was not provided"
+fi
+
+if [ -z "${BUF_TOKEN}" ]; then
+  fail "a buf authentication token was not provided"
+fi
+
+if [ -z "${BUF_INPUT}" ]; then
+  fail "an input was not provided"
+fi
+
+NOT_INSTALLED_MESSAGE='buf is not installed; please add the "bufbuild/buf-setup-action" step to your job found at https://github.com/bufbuild/buf-setup-action'
+
+BUF_COMMAND="$(type -P buf)" ||   fail "$NOT_INSTALLED_MESSAGE"
+
+if [ -z "$BUF_COMMAND" ]; then
+  fail "$NOT_INSTALLED_MESSAGE"
+fi
+
+export BUF_TOKEN
+
+"${BUF_COMMAND}" push --tag "${GITHUB_SHA}" "${BUF_INPUT}"
```

Comment:
Should we just provide `BUF_TOKEN` in the `buf push` invocation rather than exporting it? Something along the lines of:

```suggestion
BUF_TOKEN=${BUF_TOKEN} "${BUF_COMMAND}" push --tag "${GITHUB_SHA}" "${BUF_INPUT}"
```

Otherwise it will be included in all commands that follow (if we ever add any).

Reply:
I pulled in this change, but it is also unnecessary. I added `export BUF_TOKEN` because I wasn't 100% sure that environment variables declared in the action are exported. I ran a test, and it turns out that BUF_TOKEN is already exported.